mapiproxy/servers/default/emsmdb/oxcfxics.c

```diff
@@ -2570,12 +2581,20 @@ _PUBLIC_ enum MAPISTATUS EcDoRpc_RopSyncImportDeletes(TALLOC_CTX *mem_ctx,
 	owner = emsmdbp_get_owner(synccontext_object);
 	openchangedb_get_MailboxReplica(emsmdbp_ctx->oc_ctx, owner, &repl_id, &replica_guid);
 
+	object_array = &request->PropertyValues.lpProps[0].value.MVbin;
+
+	object_ids = talloc_zero_array(synccontext_object, uint64_t, object_array->cValues);
+	if (!object_ids) {
+		mapi_repl->error_code = MAPI_E_NOT_ENOUGH_MEMORY;
+		goto end;
+	}
+
 	if (request->Flags & SyncImportDeletes_Hierarchy) {
-		object_array = &request->PropertyValues.lpProps[0].value.MVbin;
 		for (i = 0; i < object_array->cValues; i++) {
 			ret = oxcfxics_fmid_from_source_key(emsmdbp_ctx, owner, object_array->bin + i, &objectID);
 			if (ret == MAPISTORE_SUCCESS) {
 				emsmdbp_folder_delete(emsmdbp_ctx, synccontext_object->parent_object, objectID, 0xff);
+				object_ids[i] = objectID;
```

Comment:
We can end up with empty entries in object_ids if oxcfxics_fmid_from_source_key returns something different from MAPISTORE_SUCCESS. Is this the intended behavior?